tinyDA/chain.py

```diff
@@ -485,7 +485,8 @@ def _update_error_model(self):
             self.posterior_coarse.likelihood.set_bias(
                 self.model_diff, self.bias.get_sigma()
             )
-
+        else:
+            raise NameError("Adaptive error model can only be state-dependent, state-independent or None.")
```

Comment:
I think this should be a `ValueError`, `NameError` is for when a [variable can't be found](https://docs.python.org/3/library/exceptions.html#NameError).

Reply:
Also, can we move this to the `__init__` method? I don't think we need to check this at every iteration.